pkg/plugin/integration_test.go

```diff
@@ -351,27 +315,256 @@ func TestCreateThenUpdateCausesNoRotations(t *testing.T) {
+				cs.Config.Images.WebConsole = "newImage"
+				perr := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az))
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "rotate cluster secrets",
+			expectMasterRotation:  true,
+			expectSyncRotation:    true,
+			expectComputeRotation: false,
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				perr := p.RotateClusterSecrets(ctx, cs, getFakeDeployer(log, cs, az))
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "readonly actions cause no rotations",
+			expectMasterRotation:  false,
+			expectSyncRotation:    false,
+			expectComputeRotation: false,
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				p.GetPluginVersion(ctx)
+				_, perr := p.ListClusterVMs(ctx, cs)
+				if perr != nil {
+					return perr
+				}
+				_, perr = p.ListEtcdBackups(ctx, cs)
+				if perr != nil {
+					return perr
+				}
+				_, perr = p.GetControlPlanePods(ctx, cs)
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "runcommand - no rotations and correct call",
+			expectMasterRotation:  false,
+			expectSyncRotation:    false,
+			expectComputeRotation: false,
+			expectCalls:           []string{"VirtualMachineScaleSetVMsClient:RunCommand:ss-master:1"},
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				perr := p.RunCommand(ctx, cs, "master-000001", "RestartDocker")
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			log := logrus.NewEntry(logrus.StandardLogger())
+			ctx := context.Background()
+			bKey, _ := tls.PrivateKeyAsBytes(testtls.DummyPrivateKey)
+			bCert, _ := tls.CertAsBytes(testtls.DummyCertificate)
+			secret := "PRIVATE KEY\n" + string(bKey) + "CERTIFICATE\n" + string(bCert)
+			secrets := []keyvault.SecretBundle{
+				{
+					ID:    to.StringPtr("PublicHostname"),
+					Value: to.StringPtr(secret),
+				},
+				{
+					ID:    to.StringPtr("Router"),
+					Value: to.StringPtr(secret),
+				},
+			}
+
+			// setup a "new" cluster
+			az := fakecloud.NewFakeAzureCloud(log, secrets)
+			p, _, err := setupNewCluster(ctx, log, cs, az)
+
+			// read the clean update blob
+			bsc := az.StorageClient.GetBlobService()
+			updateBlobService := updateblob.NewBlobService(bsc)
+			beforeBlob, err := updateBlobService.Read()
+			// get sync deployment checksum annotation
+			// FIXME: only doing it this way as fake kube Get on the deployment
+			// always returns an empty string on the annotation
+			syncer, err := sync.New(log, cs, false)
+			if err != nil {
```

Comment:
@jim-minter any better ideas on how to prove that we got (or didn't get) a sync pod rotation?